dcc-portal-server/src/main/java/org/icgc/dcc/portal/server/repository/FileRepository.java

```diff
@@ -456,6 +405,79 @@ public SearchResponse prepareSearchScroll(@NonNull String scrollId) {
         .execute().actionGet();
   }
 
+  /*
+   * Facets that aren't visible in the UI, mostly used by the Manifest Download modal dialog. These use special filters,
+   * which do not exclude self.
+   */
+  private static List<AggregationBuilder<?>> aggs() {
+    val result = ImmutableList.<AggregationBuilder<?>> builder();
+
+    val repoNameFieldName = toRawFieldName(Fields.REPO_NAME);
+
+    // repositorySize
+    val repoSizeAggKey = CustomAggregationKeys.REPO_SIZE;
+    val repoSizeTermsSubAgg = terms(repoSizeAggKey).size(MAX_FACET_TERM_COUNT).field(repoNameFieldName)
+        .subAggregation(
+            sum(CustomAggregationKeys.FILE_SIZE).field(toRawFieldName(Fields.FILE_SIZE)));
+    val repoSizeSubAgg = global(repoSizeAggKey)
+        .subAggregation(nestedAgg(repoSizeAggKey, EsFields.FILE_COPIES, repoSizeTermsSubAgg));
+    result.add(repoSizeSubAgg);
+
+    val missingSizeAgg = global(repoSizeAggKey + MISSING)
+        .subAggregation(nestedAgg(repoSizeAggKey + MISSING, EsFields.FILE_COPIES,
+            missing(repoSizeAggKey + MISSING).field(toRawFieldName(Fields.FILE_SIZE))));
+    result.add(missingSizeAgg);
+
+    return result.build();
+  }
+
+  // Special aggregation to get unique donor count for each repository
+  public TermFacet searchGroupByRepoNameDonorId(Map<String, TermFacet> facets, Query query) {
+    val facet = facets.get(CustomAggregationKeys.REPO_SIZE);
+    val terms = facet.getTerms();
+
+    if (isEmpty(terms)) {
+      return repoTermFacet(0L, 0, ImmutableList.of());
```

Comment:
That method specifically asks for an `ImmutableList`. You want me to change the signature to `List`?